### PR TITLE
feat: Report disk usage stats to metasrv thru heartbeat

### PR DESCRIPTION
**Pull request by lizhemingi**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Report region disk usage stats to metasrv through heartbeat.

Please explain IN DETAIL what the changes are in this PR and why they are needed:

- Add `region_stats` on `table` trait.
- Collect region disk usage in heartbeat.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
#1003

### CODE REVIEW

**Comment by lizhemingi** [comment]
cc @shuiyisong

**Review by sunng87 — APPROVED** [review]
LGTM

File: src/catalog/src/error.rs [hunk 1]
```diff
@@ -204,6 +204,21 @@ pub enum Error {
 
     #[snafu(display("Illegal access to catalog: {} and schema: {}", catalog, schema))]
     QueryAccessDenied { catalog: String, schema: String },
+
+    #[snafu(display(
+        "Failed to get region stats, catalog: {}, schema: {}, table: {}source: {}",
```
Comment (Fengys123) [inline src/catalog/src/error.rs:209]:
```suggestion
        "Failed to get region stats, catalog: {}, schema: {}, table: {}, source: {}",
```

**Review by shuiyisong — APPROVED** [review]
LGTM

ps: once this patch is merge, every heartbeat request would try to iterate over tables to gather region stat, which seems costing. let's figure out how to cut down the upload cost later

File: src/catalog/src/lib.rs [hunk 2]
```diff
@@ -225,10 +226,10 @@ pub(crate) async fn handle_system_table_request<'a, M: CatalogManager>(
     Ok(())
 }
 
-/// The number of regions in the datanode node.
-pub async fn region_number(catalog_manager: &CatalogManagerRef) -> Result<u64> {
-    let mut region_number: u64 = 0;
-
+/// The stat of regions in the datanode node.
+/// number of regions can be got from len of vec.
```
Comment (Fengys123) [inline src/catalog/src/lib.rs:230]:
```suggestion
/// The number of regions can be got from len of vec.
```

File: src/mito/src/table.rs [hunk 3]
```diff
@@ -331,6 +331,17 @@ impl<R: Region> Table for MitoTable<R> {
 
         Ok(())
     }
+
+    fn region_stats(&self) -> table::error::Result<Vec<RegionStat>> {
```
Comment (Fengys123) [inline src/mito/src/table.rs:335]:
```suggestion
    fn region_stats(&self) -> TableResult<Vec<RegionStat>> {
```

**Review by Fengys123 — APPROVED** [review]
LGTM. Great work, thank you @lizhemingi

**Pull request merged**